sql/core/src/test/scala/org/apache/spark/sql/execution/SparkPlanSuite.scala

```diff
@@ -97,4 +98,13 @@ class SparkPlanSuite extends QueryTest with SharedSparkSession {
       }
     }
   }
+
+  test("SPARK-34420: Throw exception if non-streaming Deduplicate is not replaced by aggregate") {
+    val df = spark.range(10)
+    val planner = spark.sessionState.planner
+    val deduplicate = Deduplicate(df.queryExecution.analyzed.output, df.queryExecution.analyzed)
+    intercept[IllegalStateException] {
+      planner.plan(deduplicate)
+    }
```

Comment:
nit: could you check the error message, too? It looks fine, otherwise.

Reply:
okay. added the check of error message.